clippy_lints/src/utils/hir.rs

```diff
@@ -100,11 +101,11 @@ impl<'a, 'tcx: 'a> SpanlessEq<'a, 'tcx> {
                 self.eq_expr(lc, rc) && self.eq_block(lt, rt) && both(le, re, |l, r| self.eq_expr(l, r))
             }
             (&ExprLit(ref l), &ExprLit(ref r)) => l.node == r.node,
-            (&ExprLoop(ref lb, ref ll, ref lls), &ExprLoop(ref rb, ref rl, ref rls)) => {
```

Comment:
Why did you remove the loop source (`{l,r}ls`)? It's just a variant-less `Copy` enum.